examples/hello-world/format-example.cpp

```diff
@@ -0,0 +1,130 @@
+
+        /// <summary>
+        /// A normal function.
+        /// </summary>
+        /// <param name="param1">The first parameter.</param>
+        /// <param name="param2">The second parameter</param>
+        /// <returns></returns>
+        int normalFunction(int param1, float* param2)
+        {
+            for (int i = 0; i < 100; i++)
+            {
+                switch (i)
+                {
+                case 0:
+                {
+                    return normalFunction(0, nullptr);
+                }
+                break;
+                case 1:
+                {
+                    float b = *param2;
+                    i--;
+                }
+                break;
+                default:
+                    break;
+                }
+            }
+            if (param1)
+                return *param2;
+            else
+                return -1;
+        }
+        class LoweredValInfo
+        {
+        };
+
+        class IRType
+        {
+        };
+
+        class IRInst
+        {
+        };
+
+        LoweredValInfo subscriptValue(IRType* type, LoweredValInfo baseVal, IRInst* indexVal)
+        {
+            auto builder = getBuilder();
+
+            // The `tryGetAddress` operation will take a complex value
+            // representation and try to turn it into a single pointer, if
+            // possible.
+            //
+            baseVal = tryGetAddress(context, baseVal, TryGetAddressMode::Aggressive);
+
+            // The `materialize` operation should ensure that we only have to
+            // deal with the small number of base cases for lowered value
+            // representations.
+            //
+            baseVal = materialize(context, baseVal);
+
+            switch (baseVal.flavor)
+            {
+            case LoweredValInfo::Flavor::Simple:
+                return LoweredValInfo::simple(
+                    builder->emitElementExtract(type, getSimpleVal(context, baseVal), indexVal));
+
+            case LoweredValInfo::Flavor::Ptr:
+                return LoweredValInfo::ptr(builder->emitElementAddress(
+                    context->irBuilder->getPtrType(type), baseVal.val, indexVal));
+
+            default:
+                SLANG_UNIMPLEMENTED_X("subscript expr");
+                UNREACHABLE_RETURN(LoweredValInfo());
+            }
+        }
+
+        int longParameterList(int param1,
+                              int param2,
+                              int param3,
+                              TestClass* param4,
+                              float* param5,
+                              std::pair<int, float> param6)
+        {
+            if (param1 == 0)
+                longParameterList(param1,
+                                  param1 + param2 + param3,
+                                  param2 - param1 + (1 << 23) / sizeof(TestClass),
+                                  param4 + (param1) * sizeof(unsigned),
+                                  param6);
```

Comment:
These bits are the main place where I'd object to the indentation choices in this example. I *think* I want `AlignAfterOpenBracket: AlwaysBreak`). Aligning all the parameters to the opening `(` has the problem that a find-replace of a function name results in changing the indent of all the parameters at all call sites.

In terms of my personal style, I'd actually like to have some of the other `Align*` settings turned on (e.g., `AlignConsecutiveAssignments`) because I think they lead to much more readable code, but I'm nervous about enabling this globally for similar reasons. I *think* the consequences of changing a variable name (and thus re-indenting its peers) are more local in practice than the consequences of changing a function name, but I'm not sure how that will work in practice.

Reply:
Added `AlignAfterOpenBracket: AlwaysBreak`